core/governance/market.go

```diff
@@ -615,6 +620,24 @@ func validateNewMarketChange(
 	return types.ProposalErrorUnspecified, nil
 }
 
+func validateUseOfEthOracles(terms *types.NewMarket, netp NetParams) (types.ProposalError, error) {
+	if terms.Changes.Instrument.Product == nil {
+		return types.ProposalErrorNoProduct, ErrMissingProduct
+	}
+
+	ethOracleEnabled, _ := netp.GetInt(netparams.EthereumOraclesEnabled)
+
+	switch product := terms.Changes.Instrument.Product.(type) {
+	case *types.InstrumentConfigurationFuture:
+		terminatedWithEthOracle := !product.Future.DataSourceSpecForTradingTermination.GetEthCallSpec().IsZero()
+		settledWithEthOracle := !product.Future.DataSourceSpecForSettlementData.GetEthCallSpec().IsZero()
+		if (terminatedWithEthOracle || settledWithEthOracle) && ethOracleEnabled != 1 {
+			return types.ProposalErrorUnsupportedProduct, ErrEthOraclesDisabled
```

Comment:
Should the error here be something like "Eth Oracle Disabled" to make it clearer why validation failed  (I imagine this will only happen a few times but still...)

Reply:
err, it is right? ```ErrEthOraclesDisabled = errors.New("ethereum oracles are disabled by network parameter")```